### PR TITLE
AB#113 increased health regeneration

### PR DESCRIPTION
**Pull request by mailaenderli**
Implemented the healthRegenBoost PowerUp. Maybe we could extract some common PowerUp code, but I would need help to achieve that.

### CODE REVIEW

File: game-backend/src/game-engine/mechanics/powerup-spawner/powerups/healthRegenBoost.test.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
+import { State, GameObjectTypes, UserContext } from '@darwin/types';
+import StateBuilder from '../../../../test-helper/StateBuilder';
+import { getUnit } from '../../../../helper/gameObjects';
+import { getConsumable } from '../../../../helper/consumable';
+import { ATTRIBUTE_BOUNDARIES } from '../../../../helper/attribute';
+import consumeHealthRegenBoost from './healthRegenBoost';
+
+describe('enduranceBoost powerup consumption', () => {
```
Comment (sir-marc) [inline game-backend/src/game-engine/mechanics/powerup-spawner/powerups/healthRegenBoost.test.ts]: You probably meant 'healthRegenboost'

File: game-backend/src/game-engine/mechanics/powerup-spawner/powerups/healthRegenBoost.test.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+    const unit = getUnit(newState, NORMAL_UNIT_ID);
+    const powerup = getConsumable(newState, POWERUP_ID);
+
+    expect(unit.attributes.healthRegenBoost).toBe(10);
+    expect(powerup).toBeUndefined();
+    expect(newState.objectIds).not.toContain(POWERUP_ID);
+  });
+
+  it('should respect attribute boundaries and not remove powerup', () => {
```
Comment (ferinator) [inline game-backend/src/game-engine/mechanics/powerup-spawner/powerups/healthRegenBoost.test.ts:58]: in the test descriptions, one time is written 'remove powerup' and the second time is written 'not remove powerup'., yet in both test cases you check for: expect(powerup).toBeUndefined(); ?

**Review by sir-marc — COMMENTED** [review]
Only one small copy pasta error, other than that, it looks great!
Great Job!

If you are interested in doing a pair programming session for refactoring and extracting some common logic, text me.

File: game-backend/src/game-engine/mechanics/powerup-spawner/powerups/healthRegenBoost.test.ts (hunk 2 above)
Comment (sir-marc) [reply game-backend/src/game-engine/mechanics/powerup-spawner/powerups/healthRegenBoost.test.ts:58]: i see `toBeDefined` on line 71

Comment (mailaenderli) [reply game-backend/src/game-engine/mechanics/powerup-spawner/powerups/healthRegenBoost.test.ts:58]: That's right but the test is `expect(powerup).toBeDefined();` and `expect(powerup).toBe`**Un**`defined();` so it should be fine?

File: game-backend/src/game-engine/mechanics/powerup-spawner/powerups/healthRegenBoost.test.ts (hunk 1 above)
Comment (mailaenderli) [reply game-backend/src/game-engine/mechanics/powerup-spawner/powerups/healthRegenBoost.test.ts]: yep of course

File: game-backend/src/game-engine/mechanics/powerup-spawner/powerups/healthRegenBoost.test.ts (hunk 2 above)
Comment (ferinator) [reply game-backend/src/game-engine/mechanics/powerup-spawner/powerups/healthRegenBoost.test.ts:58]: yes i was wrong (facepalm emoji)

**Review by StuckiSimon — APPROVED** [review]
Looks good to me, as @sir-marc mentioned, maybe you could invest some time to extract common logic in order to make the next powerups even easier to implement :v:

**Pull request merged**